src/utils/number.rs

```diff
@@ -61,71 +58,70 @@ pub(crate) fn eat_number<I: Iterator<Item = Token>>(
         });
     }
 
-    let mut dec = String::new();
-
     let next_tok = *toks.peek().unwrap();
 
-    if next_tok.kind == '.' {
+    let dec_len = if next_tok.kind == '.' {
         toks.next();
-        eat_whole_number(toks, &mut dec);
 
+        let dec = eat_whole_number(toks);
         if dec.is_empty() {
             return Err(("Expected digit.", next_tok.pos()).into());
         }
-    }
+
+        whole.push_str(&dec);
+
+        dec.len()
+    } else {
+        0
+    };
 
     let mut times_ten = String::new();
     let mut times_ten_is_postive = true;
-    #[allow(clippy::never_loop)]
-    loop {
-        if let Some(Token { kind: 'e', .. }) | Some(Token { kind: 'E', .. }) = toks.peek() {
-            let t = if let Some(tok) = toks.peek_forward(1) {
-                *tok
-            } else {
-                break;
-            };
-
-            match t.kind {
-                '-' => {
-                    toks.next();
-                    times_ten_is_postive = false;
-                }
-                '0'..='9' => {}
-                _ => break,
-            }
+    if let Some(Token { kind: 'e', .. }) | Some(Token { kind: 'E', .. }) = toks.peek() {
+        if let Some(&tok) = toks.peek() {
```

Comment:
I think this line is the reason for test failures. It should be `toks.peek_forward(1)` since we need to lookahead more than 1 character. In this case, `toks.peek()` will always return either `'e'` or `'E'`. We will also need to add back `toks.reset_cursor();	`

Reply:
Yes, I just noticed that since I tried swapping `peek` to use `itertools` but I forgot to change it back.